addons/mail/tests/discuss/test_discuss_channel_member.py

```diff
@@ -28,133 +28,32 @@ def setUpClass(cls):
-        """Test access on group."""
-        res = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertFalse(res)
-
-        # User 1 can join group with SUDO
-        self.group.with_user(self.user_1).sudo()._add_members(users=self.user_1)
-        res = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id)])
-        self.assertEqual(res.partner_id, self.user_1.partner_id)
-
-        # User 2 can not join group
-        with self.assertRaises(AccessError):
-            self.group.with_user(self.user_2)._add_members(users=self.user_2)
-
-        # User 2 can not create a `discuss.channel.member` to join the group
-        with self.assertRaises(AccessError):
-            self.env['discuss.channel.member'].with_user(self.user_2).create({
-                'partner_id': self.user_2.partner_id.id,
-                'channel_id': self.group.id,
-            })
-
-        # User 2 can not write on `discuss.channel.member` to join the group
-        channel_member = self.env['discuss.channel.member'].with_user(self.user_2).search([('is_self', '=', True)])[0]
-        with self.assertRaises(AccessError):
-            channel_member.channel_id = self.group.id
-        with self.assertRaises(AccessError):
-            channel_member.write({'channel_id': self.group.id})
-
-        # Even with SUDO, channel_id of channel.member should not be changed.
-        with self.assertRaises(AccessError):
-            channel_member.sudo().channel_id = self.group.id
-
-        # User 2 can not write on the `partner_id` of `discuss.channel.member`
-        # of an other partner to join a group
-        channel_member_1 = self.env['discuss.channel.member'].search([('channel_id', '=', self.group.id), ('partner_id', '=', self.user_1.partner_id.id)])
-        with self.assertRaises(AccessError):
-            channel_member_1.with_user(self.user_2).partner_id = self.user_2.partner_id
-        self.assertEqual(channel_member_1.partner_id, self.user_1.partner_id)
-
-        # Even with SUDO, partner_id of channel.member should not be changed.
+        cls.group = cls.env["discuss.channel"].create({"name": "Group", "channel_type": "group"})
+        cls.group.channel_member_ids.unlink()
+
+    def test_cannot_change_member_immutable_fields(self):
+        channel = self.env["discuss.channel"]._create_channel(group_id=None, name="General")
+        bob = new_test_user(self.env, "bob", groups="base.group_user")
+        another_channel = self.env["discuss.channel"]._create_channel(group_id=None, name="Another channel")
+        another_partner = self.env["res.partner"].create({"name": "John"})
+        guest = self.env["mail.guest"].create({"name": "Jane"})
+        member = channel._add_members(users=bob)
         with self.assertRaises(AccessError):
-            channel_member_1.with_user(self.user_2).sudo().partner_id = self.user_2.partner_id
-
-    def test_group_members(self):
```

Comment:
The test covered the following scenarios:
- non-member cannot invite another user to the group. Better coverage by `test_10_discuss_channel_member_access`. Specifically: 
    - `("public", "group", "outside", "other", "create", False)`
    - `("public", "group", "member", "other", "create", False)`
    - `("portal", "group", "outside", "other", "create", False)`
    - `("user", "group", "outside", "self", "create", False)`
- members can invite other users to the group. Better coverage in the same test. Specifically: `("user", "group", "member", "other", "create", True)`.
- members cannot write on other users: `("user", "group", "member", "other", "write", False)`.